web/app/js/components/Tap.jsx

```diff
@@ -0,0 +1,151 @@
+import _ from 'lodash';
+import ErrorBanner from './ErrorBanner.jsx';
+import PageHeader from './PageHeader.jsx';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { withContext } from './util/AppContext.jsx';
+import { Button, Form, Icon, Input } from 'antd';
+import './../../css/tap.css';
+
+class Tap extends React.Component {
+  static propTypes = {
+    pathPrefix: PropTypes.string.isRequired
+  }
+
+  constructor(props) {
+    super(props);
+    this.loadFromServer = this.loadFromServer.bind(this);
+
+    this.state = {
+      errors: "",
+      pollingInterval: 2000,
+      resource: "",
+      namespace: "",
+      messages: [],
+      maxLinesToDisplay: 40,
+      websocketRequestSent: false,
+      tapUnderway: false
+    };
+  }
+
+  componentWillUnmount() {
+    this.stopTapPolling();
+  }
+
+  onWebsocketRecv = e => {
+    this.setState({
+      messages: _(this.state.messages)
+        .push(e.data)
+        .takeRight(this.state.maxLinesToDisplay)
+        .value()
+    });
+  }
+
+  onWebsocketClose = e => {
+    if (!e.wasClean) {
+      this.setState({
+        errors: `Websocket [${e.code}] ${e.reason}`
+      });
+    }
+    this.stopTapPolling();
+  }
+
+  startTapPolling() {
+    this.setState({
+      messages: [],
+      tapUnderway: true
+    });
+
+    let tapWebSocket = `ws://${window.location.host}${this.props.pathPrefix}/api/tap`;
```

Comment:
Is the dashboard never served over TLS? This could use `ws` or `wss` depending on `window.location.protocol`, yea?